src/pages/distribution/component.jsx

```diff
@@ -1,30 +1,78 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import PropTypes from 'prop-types';
 import Map from 'components/map';
 import LayerToggle from 'components/map/controls/layer-toggle';
 import Icon from 'components/icon';
+import { vectorLayerCarto } from 'layers';
+import { useRouteMatch } from 'react-router-dom';
+import RampLegend from 'components/ramp-legend';
+
+import { LayerManager, Layer } from 'layer-manager/dist/components';
+import { PluginMapboxGl } from 'layer-manager';
 
 import styles from './styles.module.scss';
 
-const DistributionPageComponent = ({ viewport, setViewport, zoomIn, zoomOut }) => {
+const DistributionPageComponent = ({
+  viewport,
+  setViewport,
+  zoomIn,
+  zoomOut,
+  speciesListVisible,
+  activeSpecies
+}) => {
+  const match = useRouteMatch('/:iso');
+  const { iso } = (match && match.params) || '';
+
+  const [opacity, setOpacity] = useState(1);
+  const cartoLayer = vectorLayerCarto(iso, opacity);
+  // put active layers in the url
+  // along with its opacities
+  const layers = [cartoLayer];
+
+  const activeLayers = useMemo(() => layers.map(l => ({ ...l, active: true })), [layers]);
```

Comment:
`vectorLayerCarto` is always returning a new object so the memo is never used. You need to memo `cartoLayer` based on opacity to achieve what you want.

Reply:
will do